src/core_modules/capture-core/components/DataEntries/EnrollmentRegistrationEntry/EnrollmentRegistrationEntry.epics.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+import { ofType } from 'redux-observable';
+import { pluck, switchMap } from 'rxjs/operators';
+import { empty, from } from 'rxjs';
+import { errorCreator } from 'capture-core-utils';
+import log from 'loglevel';
+import i18n from '@dhis2/d2-i18n';
+import { enrollmentRegistrationEntryActionTypes } from './EnrollmentRegistrationEntry.actions';
+import { openDataEntryForNewEnrollmentBatchAsync } from '../Enrollment';
+import { getTrackerProgramThrowIfNotFound } from '../../../metaData/helpers';
+import { openDataEntryFailed } from '../../Pages/NewRelationship/RegisterTei/DataEntry/RegisterTeiDataEntry.actions';
+import { type TrackerProgram } from '../../../metaData/Program';
+
+export const startNewEnrollmentDataEntrySelfInitialisationEpic = (action$: InputObservable) =>
+    action$.pipe(
+        ofType(enrollmentRegistrationEntryActionTypes.TRACKER_PROGRAM_REGISTRATION_ENTRY_INITIALISATION_START),
+        pluck('payload'),
+        switchMap(({ selectedOrgUnitId, selectedScopeId: programId, dataEntryId, formFoundation }) => {
+            if (selectedOrgUnitId) {
+                let trackerProgram: ?TrackerProgram;
+                try {
+                    trackerProgram = getTrackerProgramThrowIfNotFound(programId);
+                } catch (error) {
```

Comment:
This is of course good practice, but if this fails then something is so seriously wrong that in my opinion it is ok to not catch it (I used to do exactly what you do here). I'm just saying that I will continue to be pragmatic when I'm using for example `getTrackerProgramThrowIfNotFound` and not spend time writing these. I won't complaint that you do it of course, but now you know why you won't see me do it :)

Reply:
This is basically your code :) This is why it is there. I would prefer to remove it from here. We have the error boundary from the app platform now and it will catch it. You mind if I remove it?

Reply:
This is not a biggie for me so do whatever you feel is best here :)